Danki2/Assets/Scripts/Actor/Services/MovementManager.cs

```diff
@@ -0,0 +1,60 @@
+﻿using UnityEngine;
+
+public class MovementManager
+{
+    private Actor _actor;
+    private Rigidbody _rigidbody;
+    private float _moveLockRemainingDuration;
+    private float _moveLockSpeed;
+    private bool _moveLockOn;
+    private Vector3 _moveLockDirection;
+    private Vector3 _moveDirection;
+
+    public MovementManager(Actor actor)
+    {
+        _actor = actor;
+        _rigidbody = actor.gameObject.GetComponent<Rigidbody>();
```

Comment:
I think the rigid body should also be passed in as a constructor parameter as it feels like the MovementManager shouldn't really know how to get this. Also, we want to move most of our initialisation into Awake rather than Start (we made a task for this yesterday), but this kind of thing here should only be done in Start, so having the actor be responsible for getting the rigid body should mean that the instantiation of the movement manager will stay in Start as it should

Reply:
Glad you liked it, dash should be childsplay with this branch in. You're right about the rigidbody parameter,

Reply:
I've decided **not** to make a `_rigidBody` private variable on `Actor` because I'm hoping only the`MovementManager` will need it, and I don't want people to be tempted into interacting with it directly on the `Actor`.

Reply:
Yeah that sounds sensible